server/src/main/java/io/druid/segment/loading/OmniSegmentLoader.java

```diff
@@ -52,12 +54,14 @@
   public OmniSegmentLoader(
```

Comment:
Hrm, I was actually imagining this going away.  The job of the OmniSegmentLoader was essentially to do the polymorphism that we should be getting out of `LoadSpec`s now.  It does some extra stuff, but that should be able to be lifted to the place that uses the LoadSpec.

That is, when the metadata gets deserialized, that can have a LoadSpec object in it which is just called to load the given segment.